app/controllers/admin/works_controller.rb

```diff
@@ -12,16 +12,101 @@ class Admin::WorksController < AdminController
            :remove_searchable_transcript_source, :create_combined_audio_derivatives, :update_oh_available_by_request,
            :update_oral_history_content]
 
+  before_action :sort_link_maker, only: [:index]
+
   # GET /admin/works
   # GET /admin/works.json
   def index
     # No authorize! call here. We're assuming if you can view the
     # index, you can see all published and unpublished works.
-    @q = ransack_object
-    @works = index_work_search(@q)
+    params = index_params
+
+    scope = Work
```

Comment:
Do you think it would be tidier to extract all of this stuff to like a `@works = build_search(some_args)`?